### PR TITLE
WIP: Make Brotli support enabled by default, if appropriate urllib3 version is available

### PR DESCRIPTION
**Pull request by gdubicki**
This is still a work in progress - I have to make sure that the tests are actually running, not skipped every time 😅 but [I would like to get feedback if the direction is right](https://github.com/psf/requests/blob/master/docs/dev/contributing.rst#get-early-feedback).

I had to update some deps to make the tests pass after a lot of packages being updated after I added `brotli` to dev-packages. But I assume that this should be done anyway. Especially as there were discrepancies in the versions - f.e. `pytest` version in `setup.py` commit message declared v. 4 compatibility while the version in `Pipenv` was set to be at most v. 3.10.1.

Speaking of dependencies I have a doubt how to declare required urllib3 version. Technically the first with brotli support is 1.25.1, but this is a version not supported by Requests... Should I update all the references to require urllib3 >= 1.25.2 then instead of >= 1.25.1?

### CODE REVIEW

File: docs/community/faq.rst [hunk 1]
```diff
@@ -91,3 +91,6 @@ For information on using SNI with Requests on Python < 2.7.9 refer to this
 .. _`Server-Name-Indication`: https://en.wikipedia.org/wiki/Server_Name_Indication
 .. _`virtual hosting`: https://en.wikipedia.org/wiki/Virtual_hosting
 .. _`Stack Overflow answer`: https://stackoverflow.com/questions/18578439/using-requests-with-tls-doesnt-give-sni-support/18579484#18579484
+
+
+  [#] for Brotli support you need to have `urllib3` >= 1.2.1 and `brotli` packages installed
```
Comment (gdubicki) [inline docs/community/faq.rst]: Oups, typo: 1.2.1->1.25.1.

The same in quickstart.

Comment (gdubicki) [reply docs/community/faq.rst]: I will correct it asap, of course.

**Comment by gdubicki** [comment]
Ping! :)

**Comment by sethmlarson** [comment]
FYI Requests already supports Brotli [via urllib3](https://urllib3.readthedocs.io/en/latest/advanced-usage.html#brotli-encoding):

```python
import requests

http = requests.Session()
http.headers = {"Accept-Encoding": "br"}

resp = http.request("GET", "https://cloudflare.com")
print(resp.text)
print("Content-Encoding:", resp.headers["Content-Encoding"])
```

**Comment by gdubicki** [comment]
@sethmlarson : yes, I know. Please see the updated PR title. I hope my intent is more clear now. :)

**Comment by dilyanpalauzov** [comment]
At https://github.com/psf/requests/pull/5783 I took a different approach: rely on urllib3 to determine whether to include `,br` in the `Accept-Encoding` header.

**Comment by gdubicki** [comment]
Closing in favor of #5783.

**Pull request closed without merging**